technical-reports/color/color-type.md

```diff
@@ -1,22 +1,503 @@
-# The color type
+# Color type
 
-Represents a 24bit RGB or 24+8bit RGBA color in the sRGB color space. The `$type` property MUST be set to the string `color`. The value MUST be a string containing a hex triplet/quartet including the preceding `#` character. To support other color spaces, such as HSL, translation tools SHOULD convert color tokens to the equivalent value as needed.
+Represents a color.
 
-For example, initially the color tokens MAY be defined as such:
+## Format
+
+For color tokens, the `$type` property MUST be set to the string `color`.
+
+The `$value` property can then be used to specify the details of the color, The `$value` object contains the following properties:
+
+- `colorSpace` (required): A string that specifices the color space. For supported color spaces, see the [supported color spaces](#supported-color-spaces) section below.
+- `components` (required): An array representing the color components. The number of components depends on the color space. Each element of the array MUST be either:
+  - A number
+  - The 'none' keyword
+- `alpha` (optional): A number that represents the alpha value of the color. This value is between `0` and `1`, where `0` is fully transparent and `1` is fully opaque.
+- `fallback` (optional): A string that represents a fallback value of the color in [CSS hex color notation](https://www.w3.org/TR/css-color-4/#hex-notation) format. The fallback color MUST omit the alpha value, since alpha is specified in the `alpha` property. The fallback value is used when the color cannot be represented in the specified color space.
 
 <aside class="example">
 
 ```json
 {
-  "Majestic magenta": {
-    "$value": "#ff00ff",
-    "$type": "color"
+  "Hot pink": {
+    "$type": "color",
+    "$value": {
+      "colorSpace": "srgb",
+      "components": [255, 0, 255],
```

Comment:
```suggestion
      "components": [1, 0, 1],
```

Reply:
sidenote: this is a really clear example! Seeing how `[1, 0, 1]` relates to `#ff00ff` for most folks will make it “click” I think (even if they’re not familiar with CSS Color Module 4)